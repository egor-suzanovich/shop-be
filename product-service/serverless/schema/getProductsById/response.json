{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "properties": {
    "count": {
      "type": "number"
    },
    "description": {
      "type": "string"
    },
    "id": {
      "type": "string"
    },
    "price": {
      "type": "number"
    },
    "title": {
      "type": "string"
    }
  },
  "type": "object"
}
