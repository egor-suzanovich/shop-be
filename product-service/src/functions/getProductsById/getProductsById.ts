import {
  APIGatewayProxyEvent,
  APIGatewayProxyResult,
  APIGatewayProxyEventPathParameters,
} from 'aws-lambda';
import {
  getApiErrorResponse,
  getApiResponse,
  getProductNotFoundError,
} from '../../helpers/apiHelpers';
import { getProductsList } from '../../helpers/products';
import commonMidlware from '../../lib/commonMidlware';
import { ProductPathParameter, ProductsListItem } from '../types';

export const originalHandler = async (
  event: APIGatewayProxyEvent,
): Promise<APIGatewayProxyResult> => {
  const { productId } = event.pathParameters as ProductPathParameter;
  
  console.log(`[Incoming productId]: ${productId}`);

  try {
    const productsList = (await getProductsList()) as Array<ProductsListItem>;
    const product = productsList.find((p) => p.id === productId);

    if (!product) {
      return getProductNotFoundError(`Product with ID: "${productId}" not found`);
    }

    return getApiResponse(product);
  } catch (error) {
    console.log(error);
    return getApiErrorResponse();
  }
};

export const handler = commonMidlware(originalHandler);
