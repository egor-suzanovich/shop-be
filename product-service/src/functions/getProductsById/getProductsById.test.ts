import { describe, expect } from '@jest/globals';
import { originalHandler } from './getProductsById';

describe('getProductsById tests', () => {
  const mockEvent = {
    pathParameters: {
      productId: '7567ec4b-b10c-48c5-9345-fc73c48a80a1',
    },
  } as any;

  it('should return valid response', async () => {
    const response = await originalHandler(mockEvent);
    expect(response).toMatchSnapshot();
  });
});
