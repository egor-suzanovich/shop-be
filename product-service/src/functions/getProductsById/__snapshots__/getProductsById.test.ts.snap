// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProductsById tests should return valid response 1`] = `
Object {
  "body": "{\\"count\\":12,\\"description\\":\\"Short Product Description7\\",\\"id\\":\\"7567ec4b-b10c-48c5-9345-fc73c48a80a1\\",\\"price\\":15,\\"title\\":\\"ProductTitle\\"}",
  "headers": Object {
    "Access-Control-Allow-Headers": "*",
    "Access-Control-Allow-Origin": "*",
    "Cache-Control": "no-cache, no-store, max-age=0",
    "Content-Type": "application/json",
  },
  "statusCode": 200,
}
`;
